pypeit/edgetrace.py

```diff
@@ -4195,87 +4237,109 @@ def maskdesign_matching(self, debug=False):
+                                                         self.spectrograph.slitmask.slitid[ind_b], strict=True))
+            ind_b = ind_b[sortind_b]
+            # define which side to add the trace and insert it
+            lside = -np.ones(bot_edge_pred[needind_b].shape[0], dtype=int)
+            missing_left_traces = self.predict_traces(bot_edge_pred[needind_b], side=lside)
+            self.insert_traces(lside, missing_left_traces, mode='mask')
+
+        if needind_t.size > 0:
+            msgs.info('Adding {} right missing edge(s)'.format(needind_t.size))
+            # Append the missing indices and re-sort all
+            ind_t = np.append(ind_t, needind_t)
+            sortind_t = np.argsort(utils.index_of_x_eq_y(self.spectrograph.slitmask.slitid[sortindx],
+                                                         self.spectrograph.slitmask.slitid[ind_t],strict=True))
+            ind_t = ind_t[sortind_t]
+            # define which side to add the trace and insert it
+            rside = np.ones(top_edge_pred[needind_t].shape[0], dtype=int)
+            missing_right_traces = self.predict_traces(top_edge_pred[needind_t], side=rside)
+            self.insert_traces(rside, missing_right_traces, mode='mask')
+
+        if debug:
+            if (needind_b.size > 0) | np.any(dupl_b):
+                slitdesign_matching.plot_matches(self.edge_fit[:, self.is_left], ind_b, bot_edge_pred, reference_row,
+                                                 self.spectrograph.slitmask.slitindx, nspat=self.nspat, edge='left')
+            if (needind_t.size > 0) | np.any(dupl_t):
+                slitdesign_matching.plot_matches(self.edge_fit[:, self.is_right], ind_t, top_edge_pred, reference_row,
+                                                 self.spectrograph.slitmask.slitindx, nspat=self.nspat, edge='right')
+
+        self.maskdef_id = np.zeros(self.ntrace, dtype=int)
+        self.maskdef_id[self.is_left] = self.spectrograph.slitmask.slitid[ind_b]
+        self.maskdef_id[self.is_right] = self.spectrograph.slitmask.slitid[ind_t]
+
+        self.align_slit = np.zeros(self.ntrace, dtype=bool)
+        self.align_slit[self.is_left] = self.spectrograph.slitmask.alignment_slit[ind_b]
+        self.align_slit[self.is_right] = self.spectrograph.slitmask.alignment_slit[ind_t]
+
+        # Propagate the coefficients, `coeff_b` and `coeff_t`, of the x-correlation and the
+        # left and right spatial position of the slit edges from optical model (before x-correlation)
+        # with the purpose to fill a table with the information on slitmask design matching. The table will
+        # be filled out at the very end of the slit tracing process, in `get_slits()`.
+        self.coeff_b = coeff_b
+        self.coeff_t = coeff_t
+        self.omodel_bspat = omodel_bspat
+        self.omodel_tspat = omodel_tspat
+
+    def _fill_design_table(self, maskdef_id, coeff_b, coeff_t, omodel_bspat, omodel_tspat):
```

Comment:
As this effectively defines the DataModel for this Table, 
kindly add details to the doc string on what is each column.

Reply:
Done.